data/store/mongo/mongo.go

```diff
@@ -544,6 +545,16 @@ func (d *DataRepository) isDatumToBucket(datum data.Datum) bool {
 	return false
 }
 
+func (d *DataRepository) isDatumToLegacy(datum data.Datum) bool {
+	datumType := datum.GetType()
+	for _, legacyType := range d.DataTypesLegacy {
```

Comment:
Just a comment/question: why not using a map here? It would be easier (more efficient I don't know...) to verify a type is in the list.